src/lib/login/LoginService.js

```diff
@@ -105,6 +110,18 @@ class LoginService {
       throw exception
     }
   }
+
+  async validateJWTExistenceAndExpiration(): Promise<string | null> {
+    const jwt = await this.getJWT()
+    if (jwt) {
+      const decoded = jsonwebtoken.decode(jwt, { json: true })
+
+      if (decoded.exp && Date.now() >= decoded.exp * 1000) {
+        return jwt
+      }
+    }
+    return null
+  }
```

Comment:
```suggestion
  async validateJWTExistenceAndExpiration(): Promise<string | null> {
    const jwt = await this.getJWT()
    const { profilePublickey } = await this.getCredentials()

    if (!jwt) {
      return null
    }
    
    const decoded = await new Promise(resolve => 
      jsonwebtoken.decode(jwt, { json: true }, (error, decoded) => resolve(error ? null : decoded))
    )       

    if (!decoded || !decoded.exp || Date.now() < decoded.exp * 1000) {
      return null
    }

    return jwt
  }
```

Reply:
also you missed up with date condition
Date.now() >= decoded.exp * 1000 means that current date is **later** that expiration. so the key is expired.
actually it should be Date.now() < decoded.exp * 1000 e.g. expiration date is later then current

in my implementation i negated the whole condition to decrease indent level, so that expression was left unchanged (as i'm checking now for **is expired**)